Mage.Sets/src/mage/cards/a/AnrakyrTheTraveller.java

```diff
@@ -0,0 +1,198 @@
+package mage.cards.a;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import mage.ApprovingObject;
+import mage.MageInt;
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.ActivatedAbility;
+import mage.abilities.PlayLandAbility;
+import mage.abilities.SpellAbility;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.Costs;
+import mage.abilities.costs.CostsImpl;
+import mage.abilities.costs.common.PayLifeCost;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.cost.CastFromHandForFreeEffect;
+import mage.cards.AdventureCard;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.CardWithHalves;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.Zone;
+import mage.filter.FilterCard;
+import mage.filter.FilterSpell;
+import mage.filter.common.FilterArtifactCard;
+import mage.filter.common.FilterArtifactSpell;
+import mage.filter.common.FilterCreatureAttackingYou;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.util.CardUtil;
+import mage.util.CardUtil.SpellCastTracker;
+
+/**
+ * @author gravitybone
+ * @author Aquid
+ */
+public final class AnrakyrTheTraveller extends CardImpl {
+
+    private static final FilterCard filter = new FilterArtifactCard();
```

Comment:
Could move that static filter in `AnrakyrTheTravellerEffect`, as there are no variation on it.
And we probably have a `StaticFilter` for it anyway.

Reply:
Moved the filter to `AnrakyrTheTravellerEffect`, but did not use a `StaticFilter` because `CardUtils.getCastableComponents` uses a `FilterCard` but the text should be `an artifact spell` for which there is no `StaticFilter`.